nixos/modules/services/networking/shellhub-agent.nix

```diff
@@ -74,9 +79,13 @@ in {
         "time-sync.target"
       ];
 
-      environment.SERVER_ADDRESS = cfg.server;
-      environment.PRIVATE_KEY = cfg.privateKey;
-      environment.TENANT_ID = cfg.tenantId;
+      environment = {
+        SHELLHUB_SERVER_ADDRESS = cfg.server;
+        SHELLHUB_PRIVATE_KEY = cfg.privateKey;
+        SHELLHUB_TENANT_ID = cfg.tenantId;
+        SHELLHUB_KEEPALIVE_INTERVAL = toString cfg.keepAliveInterval;
+        SHELLHUB_PREFERRED_HOSTNAME = cfg.preferredHostname;
```

Comment:
Considering the name isn't this option optional? If so, maybe consider setting this value to default for `null` and using `lib.mkIf (cfg.preferredHostname != null) cfg.preferredHostname` here instead.

Reply:
It does allow it being empty; I changed it to not require it being set.